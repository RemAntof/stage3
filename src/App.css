.appContainer {
  display: flex;
  width: 100%;
  overflow: hidden;
  position: relative;
}

.mainPage {
  flex-shrink: 0;
  width: 100%;
  transition:
    transform 0.3s ease-in-out,
    width 0.3s ease-in-out;
}

#outlet {
  position: fixed;
  top: 0;
  right: 0;
  width: 0;
  height: 100%;
  transition: width 0.3s ease-in-out;
  background-color: #242424;
}

#outlet.visible {
  width: fit-content;
}

.mainPage.shifted {
  width: 60%;
}
