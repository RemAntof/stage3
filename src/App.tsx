import React, { useEffect, useState, useRef } from 'react';
import './App.css';
import MainPage from '@views/mainPage/MainPage';
import {
  Outlet,
  useLocation,
  useNavigate,
} from 'react-router-dom';

const App: React.FC = () => {
  const location = useLocation();
  const [outletVisible, setOutletVisible] = useState(false);
  const outletRef = useRef<HTMLDivElement>(null);
  const navigate = useNavigate();

  useEffect(() => {
    setOutletVisible(location.pathname !== '/');
  }, [location]);

  const onClick = (
    event: React.MouseEvent<HTMLDivElement, MouseEvent>
  ) => {
    if (
      outletVisible &&
      outletRef.current &&
      !outletRef.current.contains(event.target as Node)
    ) {
      setOutletVisible(false);
      navigate(-1);
    }
  };

  return (
    <div className="appContainer" onClick={onClick}>
      <div
        className={`mainPage ${outletVisible ? 'shifted' : ''}`}
      >
        <MainPage />
      </div>
      <div
        id="outlet"
        ref={outletRef}
        className={outletVisible ? 'visible' : 'hidden'}
      >
        {outletVisible ? <Outlet /> : null}
      </div>
    </div>
  );
};

export default App;
