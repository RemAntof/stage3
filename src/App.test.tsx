import { render, act } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import App from './App';

test('renders App component correctly', async () => {
  await act(async () => {
    render(
      <MemoryRouter>
        <App />
      </MemoryRouter>
    );
  });
});
