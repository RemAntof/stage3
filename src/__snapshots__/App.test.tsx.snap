// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component snapshot - dark theme, outlet hidden 1`] = `
<DocumentFragment>
  <div
    class="appContainer dark "
    data-testid="app-container"
  >
    <div
      class="mainPage "
    >
      <div
        class="container"
      >
        <input
          id="checkbox"
          type="checkbox"
        />
        <div
          class="background"
        >
          <img
            class="stars"
            src="https://gateway.pinata.cloud/ipfs/Qmcd1rqt5BJrQYX9hHeWc8AqEhBp2pj6Mis82FC5wVW2a2"
          />
          <div
            class="clouds"
          >
            <img
              class="stars"
              src="https://gateway.pinata.cloud/ipfs/QmdDiiwPP4VC5jmRTtPa1v4MrA5JUesq8WZHs8ACLt3Vcj"
            />
            <img
              class="stars"
              src="https://gateway.pinata.cloud/ipfs/QmYNhwZuMNCucpn8YgGxPdKBnVsBDr473pbXCmxKE4uPJa"
            />
          </div>
        </div>
        <label
          class="content"
          for="checkbox"
        >
          <div
            class="sun__moon__container"
          >
            <div
              class="sun__moon"
            >
              <div
                class="sun"
              />
              <div
                class="moon"
              >
                <div />
                <div />
                <div />
              </div>
            </div>
            <div
              class="rays"
            >
              <div />
              <div />
              <div />
            </div>
          </div>
        </label>
      </div>
      <div>
        <h1>
          Something went wrong.
        </h1>
      </div>
    </div>
    <div
      class="hidden"
      data-testid="outlet"
      id="outlet"
    />
  </div>
</DocumentFragment>
`;

exports[`App component snapshot - dark theme, outlet visible 1`] = `
<DocumentFragment>
  <div
    class="appContainer dark shifted"
    data-testid="app-container"
  >
    <div
      class="mainPage shifted"
    >
      <div
        class="container"
      >
        <input
          id="checkbox"
          type="checkbox"
        />
        <div
          class="background"
        >
          <img
            class="stars"
            src="https://gateway.pinata.cloud/ipfs/Qmcd1rqt5BJrQYX9hHeWc8AqEhBp2pj6Mis82FC5wVW2a2"
          />
          <div
            class="clouds"
          >
            <img
              class="stars"
              src="https://gateway.pinata.cloud/ipfs/QmdDiiwPP4VC5jmRTtPa1v4MrA5JUesq8WZHs8ACLt3Vcj"
            />
            <img
              class="stars"
              src="https://gateway.pinata.cloud/ipfs/QmYNhwZuMNCucpn8YgGxPdKBnVsBDr473pbXCmxKE4uPJa"
            />
          </div>
        </div>
        <label
          class="content"
          for="checkbox"
        >
          <div
            class="sun__moon__container"
          >
            <div
              class="sun__moon"
            >
              <div
                class="sun"
              />
              <div
                class="moon"
              >
                <div />
                <div />
                <div />
              </div>
            </div>
            <div
              class="rays"
            >
              <div />
              <div />
              <div />
            </div>
          </div>
        </label>
      </div>
      <div>
        <h1>
          Something went wrong.
        </h1>
      </div>
    </div>
    <div
      class="visible"
      data-testid="outlet"
      id="outlet"
    />
  </div>
</DocumentFragment>
`;

exports[`App component snapshot - light theme, outlet hidden 1`] = `
<DocumentFragment>
  <div
    class="appContainer light "
    data-testid="app-container"
  >
    <div
      class="mainPage "
    >
      <div
        class="container"
      >
        <input
          id="checkbox"
          type="checkbox"
        />
        <div
          class="background"
        >
          <img
            class="stars"
            src="https://gateway.pinata.cloud/ipfs/Qmcd1rqt5BJrQYX9hHeWc8AqEhBp2pj6Mis82FC5wVW2a2"
          />
          <div
            class="clouds"
          >
            <img
              class="stars"
              src="https://gateway.pinata.cloud/ipfs/QmdDiiwPP4VC5jmRTtPa1v4MrA5JUesq8WZHs8ACLt3Vcj"
            />
            <img
              class="stars"
              src="https://gateway.pinata.cloud/ipfs/QmYNhwZuMNCucpn8YgGxPdKBnVsBDr473pbXCmxKE4uPJa"
            />
          </div>
        </div>
        <label
          class="content"
          for="checkbox"
        >
          <div
            class="sun__moon__container"
          >
            <div
              class="sun__moon"
            >
              <div
                class="sun"
              />
              <div
                class="moon"
              >
                <div />
                <div />
                <div />
              </div>
            </div>
            <div
              class="rays"
            >
              <div />
              <div />
              <div />
            </div>
          </div>
        </label>
      </div>
      <div>
        <h1>
          Something went wrong.
        </h1>
      </div>
    </div>
    <div
      class="hidden"
      data-testid="outlet"
      id="outlet"
    />
  </div>
</DocumentFragment>
`;
